docs/sc-memory/api/cpp/extended/permissions_api.md

```diff
@@ -0,0 +1,809 @@
+*];;
+
+nrel_user_action_class_within_sc_structure
+~|> (..user => (action_read_from_sc_memory => ..structure));;
+        </code>
+      </pre>
+    </td>
+  </tr>
+  <tr>
+    <td>Each user in users set hasn't global read permissions</td>
+    <td><scg src="../images/permissions/no_global_read_permissions_for_users_set.gwf"></scg></td>
+    <td>
+      <pre>
+        <code class="js hljs javascript">
+nrel_users_set_action_class
+~|> (concept_user => action_read_from_sc_memory);;
+        </code>
+      </pre>
+    </td>
+  </tr>
+
+  <tr>
+    <td>Each user in users set hasn't local read permissions</td>
+    <td><scg src="../images/permissions/no_local_read_permissions_for_users_set.gwf"></scg></td>
+    <td>
+      <pre>
+        <code class="js hljs javascript">
+..structure = [*
+    concept_set ~> ..set;;
+*];;
+
+nrel_users_set_action_class_within_sc_structure
+~|> (..user => (action_read_from_sc_memory => ..structure));;
+        </code>
+      </pre>
+    </td>
+  </tr>
+</table>
+
+In these examples, you can use any other action class (`action_generate_in_sc_memory`, `action_erase_from_sc_memory`, `action_read_permissions_from_sc_memory`, `action_generate_permissions_in_sc_memory` or `action_erase_permissions_from_sc_memory`) instead of `action_read_from_sc_memory`.
+
+All sc-arcs from permissions classes (relations) must be permanent or temporary. All sc-arcs from user groups (classes or sets) must be permanent or temporary.
+
+!!! note
+    If the knowledge base does not explicitly specify permissions for a user, it is equivalent to saying that this user does not have them.
+
+!!! note
+    In order to erase permissions for a user it is enough to erase an access sc-arc from the relation that indicated permissions.
+  
+!!! note
+    In the sc-machine there is a global system context -- `ScMemory::ms_globalContext`, that has all permissions. You can use it to update permissions that can be used to read, update or erase permissions for other users.
+
+!!! warning
+    Erasing a positive access sc-arc between relation and sc-arc between the user and an action class, that the user can perform, automatically creates a negative access sc-arc between the given relation and the sc-arc between the user and the action class.
+
+!!! warning
+    Creating a positive access sc-arc between relation and sc-arc between the user and an action class, that the user can perform, automatically erases all negative access sc-arcs between the given relation and the sc-arc between the user and the action class.
+
+---
+
+### **Compatibility of global and local user permissions**
+
+Both global and local permissions can be described for the same user. It is important to know the logic of how they work together. For example, concrete user wants to handle some sc-construction. There may be several cases:
+
+<table>
+  <thead>
+    <tr>
+      <th>Case</th>
+      <th>Does specified sc-construction belong to any permitted sc-structure?</th>
```

Comment:
Need to discuss. For me it is not clear how to read this table

Reply:
Ok, lets discuss

Reply:
Mark result in table

Reply:
Fixed